Extension/src/LanguageServer/settings.ts

```diff
@@ -313,6 +314,7 @@ export class CppSettings extends Settings {
 
     public get maxConcurrentThreads(): number | undefined | null { return super.Section.get<number | null>("maxConcurrentThreads"); }
     public get maxMemory(): number | undefined | null { return super.Section.get<number | null>("maxMemory"); }
+    public get maxSymbolSearchResults(): number | undefined | null { return super.Section.get<number | null>("maxSymbolSearchResults"); }
```

Comment:
My preference would still be use `null` as the default, but otherwise shouldn't the `null` type be removed or is that valid still?

Reply:
`null`/`default` is what we use when we want a dynamic default based on some machine or OS properties.  I don't believe we need a dynamic default here because we don't check the machine hardware for a reason to increase/decrease the default setting value.

But yes, we should remove `null` here.